core/ibc/src/grpc.rs

```diff
@@ -780,74 +839,63 @@ impl<Ctx: ClientReader + ClientKeeper> IbcClientMsgService<Ctx> {
     pub fn new(ctx: Arc<RwLock<Ctx>>) -> Self {
         Self { ctx }
     }
+
+    fn process_msg(&self, msg: ClientMsg) -> Result<ClientResult, tonic::Status> {
+        let mut ctx = self
+            .ctx
+            .write()
+            .map_err(|v| tonic::Status::internal(format!("{:?}", v)))?;
+        let output = client_dispatch(ctx.deref(), msg)
```

Comment:
Is it necessary to use `WriteGuard` here, or ReadGuard` also works?

Reply:
read is enough for dispatch, but the following store_**_result needs Write